File: ch04/Toolbar1/Toolbar1/ViewController.m
//
//  ViewController.m
//  Toolbar1
//
//  Created by MyZiyue on 2018/12/13.
//  Copyright © 2018 MyZiyue.com. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@property (strong, nonatomic) UILabel *label;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
   
    CGRect screen = [[UIScreen mainScreen] bounds];
    
    CGFloat toolbarHeight = 44;
    
    // 1. toolbar
    UIToolbar* toolbar = [[UIToolbar alloc] initWithFrame:CGRectMake(0, screen.size.height-toolbarHeight, screen.size.width, toolbarHeight)];
    
    UIBarButtonItem *saveButtonItem = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemSave target:self action:@selector(save:)];
    UIBarButtonItem *openButtonItem = [[UIBarButtonItem alloc] initWithTitle:@"Open" style:UIBarButtonItemStylePlain target:self action:@selector(open:)];
    UIBarButtonItem *flexibleButtonItem = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemFlexibleSpace target:self action:nil];
    toolbar.items = @[saveButtonItem, flexibleButtonItem, openButtonItem];
    [self.view addSubview:toolbar];
    
    // 2. label
    CGFloat labelWidth = 84;
    CGFloat labelHeight = 21;
    CGFloat labelTopView = 250;
    
    self.label = [[UILabel alloc] initWithFrame:CGRectMake((screen.size.width - labelWidth)/2, labelTopView, labelWidth, labelHeight)];
    self.label.text = @"Label";
    // 字体左右居中
    self.label.textAlignment = NSTextAlignmentCenter;
    [self.view addSubview:self.label];
}

-(void)save:(id)sender {
    self.label.text = @"点击Save";
}

-(void)open:(id)sender {
    self.label.text = @"点击Open";
}
@end

File: ch04/SToolbar1/SToolbar1/ViewController.swift
//
//  ViewController.swift
//  SToolbar1
//
//  Created by MyZiyue on 2018/12/13.
//  Copyright © 2018 MyZiyue.com. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    var label: UILabel!

    override func viewDidLoad() {
        super.viewDidLoad()
        
        let screen = UIScreen.main.bounds
        
        let toolbarHeight:CGFloat = 44
        
        // 1. 添加Toolbar
        let toolbar = UIToolbar(frame: CGRect(x:0, y:(screen.size.height - toolbarHeight), width:screen.size.width, height:toolbarHeight))
        
        let saveButtonItem = UIBarButtonItem(barButtonSystemItem: UIBarButtonItem.SystemItem.save, target: self, action: #selector(save))
        let openButtonItem = UIBarButtonItem(title: "Open", style: UIBarButtonItem.Style.plain, target: self, action: #selector(open))
        let flexibleButtonItem = UIBarButtonItem(barButtonSystemItem: UIBarButtonItem.SystemItem.flexibleSpace, target: self, action: nil)
        toolbar.items = [saveButtonItem, flexibleButtonItem, openButtonItem]
        
        self.view.addSubview(toolbar)
        
        // 2. 添加标签
        let labelWidth: CGFloat = 84
        let labelHeight: CGFloat = 21
        let labelTopView: CGFloat = 250
        
        self.label = UILabel(frame: CGRect(x:(screen.size.width - labelWidth)/2, y:labelTopView, width:labelWidth, height:labelHeight))
        self.label.text = "Label"
        // 字体左右居中
        self.label.textAlignment = .center
        self.view.addSubview(self.label)
    }

    @objc func save(sender: Any) {
        self.label.text = "点击Save"
    }
    @objc func open(sender: Any) {
        self.label.text = "点击Open"
    }

}

